cobalt/akn.py

```diff
@@ -0,0 +1,403 @@
+"""
+Cobalt's class hierarchy mimics that of the Akoma Ntoso standard. There is a primary root class for
+all Akoma Ntoso documents. There are subclasses for each of the Akoma Ntoso document structure types,
+such as hierarchicalStructure, debateStructure, etc. Finally, there is a class for each Akoma Ntoso
+document type (act, bill, judgment, etc.) that extends the corresponding structure type.
+"""
+from collections import OrderedDict
+
+import re
+from lxml import etree
+from lxml import objectify
+from iso8601 import parse_date
+
+from .uri import FrbrUri
+
+
+ENCODING_RE = re.compile(r'encoding="[\w-]+"')
+
+DATE_FORMAT = "%Y-%m-%d"
+
+AKN_NAMESPACES = {
+    '2.0': 'http://www.akomantoso.org/2.0',
+    '3.0': 'http://docs.oasis-open.org/legaldocml/ns/akn/3.0',
+}
+
+
+def datestring(value):
+    if value is None:
+        return ""
+    elif isinstance(value, str):
+        return value
+    else:
+        return "%04d-%02d-%02d" % (value.year, value.month, value.day)
+
+
+# Create a new objectify parser that doesn't remove blank text nodes
+objectify_parser = etree.XMLParser()
+objectify_parser.set_element_class_lookup(objectify.ObjectifyElementClassLookup())
+
+
+class AkomaNtosoDocument:
+    """ Base class for Akoma Ntoso documents.
+    """
+    _parser = objectify_parser
+
+    def __init__(self, xml=None):
+        encoding = ENCODING_RE.search(xml, 0, 200)
```

Comment:
I *think* we can now throw this away if we're intelligent about how we parse strings vs bytes.

Reply:
I'm not clear on what you mean here?

Reply:
This was a bit of a hack that I think we can now handle differently. We'll get there.